protocols/swap/reports/SECURITY.md

```diff
@@ -1,175 +1,173 @@
-| swap | Swap.sol | external   | `Types.Order _order, Types.Signature _signature` | no      |
-| cancel     | Swap.sol | external   | `uint256[] calldata _nonces`                                                                                                                                                                                            | no      |
-| invalidate | Swap.sol | external   | `uint256 _minimumNonce`                                                                                                                                                                                                 | no      |
-| authorize  | Swap.sol | external   | `address _delegate`, `uint256 _expiry`                                                                                                                                                                                  | no      |
-| revoke     | Swap.sol | external   | `address _delegate`                                                                                                                                                                                                     | no      |
-
-Note that state-modifying public signal points are minimized in this contract, reducing
-the exploit surface to incorrect arguments to swap, authorize, cancel, invalidate, and revoke.
+| Function        | Source   | Visibility | Params                                         | Payable |
+| :-------------- | :------- | :--------- | :--------------------------------------------- | :------ |
+| swap            | Swap.sol | external   | `Types.Order _order`                           | no      |
+| cancel          | Swap.sol | external   | `uint256[] calldata _nonces`                   | no      |
+| invalidate      | Swap.sol | external   | `uint256 _minimumNonce`                        | no      |
+| authorizeSender | Swap.sol | external   | `address _authorizedSender`, `uint256 _expiry` | no      |
+| authorizeSigner | Swap.sol | external   | `address _authorizedSigner`, `uint256 _expiry` | no      |
+| revokeSender    | Swap.sol | external   | `address _authorizedSender`                    | no      |
+| revokeSigner    | Swap.sol | external   | `address _authorizedSender`                    | no      |
 
 ## Invariants
 
-#### 1. No ether or tokens should be held by the contract address.
+### No ether (ETH) or tokens should be held by the contract address.
 
+- No functions are payable and therefore ether (ETH) cannot be stuck by calling any functions.
+- By inspection, all branches of the function `swap` either succeed or throw. At no point can these functions return false or fail silently without throwing.
+- All transfers in the execution function are between the signer and sender or a signer and affiliate; at no point is a token transfer performed to or from the Swap contract address.
```

Comment:
Well, anyone could perform a token transfer to the swap contract... but that's their fault. And again it never uses its own balance. This is sort of the same point as the `seldestruct` point

Reply:
Oh you wrote that as the next point hahah. Okay ignore me.